web/client/components/map/openlayers/Map.jsx

```diff
@@ -228,14 +261,36 @@ export default class OpenlayersMap extends React.Component {
         if (newProps.mousePointer !== this.props.mousePointer) {
             this.setMousePointer(newProps.mousePointer);
         }
-
         if (newProps.zoomControl !== this.props.zoomControl) {
             if (newProps.zoomControl) {
                 this.map.addControl(new Zoom());
             } else {
                 this.map.removeControl(this.map.getControls().getArray().filter((ctl) => ctl instanceof Zoom)[0]);
             }
         }
+        /*
+         * Manage interactions programmatically.
+         * map interactions may change, i.e. becoming enabled or disabled
+         */
+        if (this.map && (this.props.mapOptions && this.props.mapOptions.interactions) !== (newProps.mapOptions && newProps.mapOptions.interactions)) {
+            const newInteractions = newProps.mapOptions.interactions || {};
+            Object.keys(newInteractions).forEach(newInteraction => {
+                this.map.getInteractions().getArray().forEach((mapInteraction) => {
+                    if (interactionsDefaults[newInteraction] && mapInteraction instanceof interactionsDefaults[newInteraction].Instance) {
+                        this[`${newInteraction}Interaction`] = mapInteraction;
+                    }
+                });
+                if (!this[`${newInteraction}Interaction`]) {
+                    // if the requested interaction does not exist in the map then add it
+                    this[`${newInteraction}Interaction`] = new interactionsDefaults[newInteraction].Instance(interactionsDefaults[newInteraction].options);
+                    this.map.addInteraction(this[`${newInteraction}Interaction`]);
+                } else {
+                    // otherwise use existing interaction and enable or disable it based on newProps values
+                    this[`${newInteraction}Interaction`].setActive(newInteractions[newInteraction]);
+                }
+            });
```

Comment:
strange algorithm 
 -  create new interaction and add if it wasn't there
 - if at least once in the object lifetime (see the comment before) it existed, set active, if the value is true or false. 
Can you make it simpler? Please take even into account that these interactions can have options, so don't wire the solution to your use case.

Reply:
the problem is that we should not duplicate existing interactions. so :
if (already exists) then 
   activate / deactivate  (to do this i need to fetch the existing interaction, and probably i can 
   avoid to store it in the this object)
else
  create it and addit to the map

side note: regarding the options they are mostly added with defaults and a few of them have some customization. btw i am not removing existing interactions